Restrict Risk Table 3 Lookup

mickohanlon23 opened the pull request:
**Jira Ticket**: This PR addresses **[CCSMCDS-75](https://jira.mitre.org/browse/CCSMCDS-75)**

**Description**: Only return a value from lookup table 3 if the biopsy result is the most recent result.

**Notes on Patient Testing**:
The following test patients were affected by this PR:

1. SJ42
   - Former recommendation: Treatment from Risk Table 3, because of a previous CIN2 result
   - Clinical Situation: this patient is in surveillance for CIN2 (they chose Observation as opposed to treatment)
   - Current recommendation: Colposcopy from Errata
   - Analysis: Makes sense, since results from follow up cotesting during observation of CIN2 would lead to this patient getting another colpo

2. PN33
   - Former recommendation: Treatment from Risk Table 3, because of a previous CIN2 result
   - Clinical Situation: this patient is in surveillance for CIN2 (they chose Observation as opposed to treatment)
   - Current recommendation: Colposcopy from Risk Table 1
   - Analysis: I believe this makes sense, since the persistent ASC-US / HPV+ cotest after the CIN2 would lead this patient to get another colposcopy during Observation.

3. MT53
   - Former recommendation: 1-year-surveillance from risk table 3, because of former CIN1 result.
   - Clinical Situation: this patient is in surveillance after a low-grade biopsy
   -  Current recommendation: Unknown, since this case was discovered with the pilot partners after our mock test cases were created
   - Analysis: I believe this patient would now receive a colposcopy rec from the errata, due to subsequent ASC-US / HPV+ cotest results during surveillance of CIN1. May also come from section I, or risk table 4, however, either way, should be a colposcopy recommendation.

mrnosal commented on cql/ManageCommonAbnormality.cql:
```diff
@@ -482,7 +488,7 @@ define ErrataRecommendation:
       AgeInYears() >= 25 and
       CytologyAloneInterpretedAsAscus and
-      Collate.MostRecentReport = MostRecentCytologyAlone.date and
+      Collate.HasNoBiopsyOrTreatmentAfterDate(MostRecentCytologyAlone.date) and
```
This is a duplicate of the next line? Did you mean to remove or change the last line?

mickohanlon23 replied:
That was supposed to be `IsMostRecentTest`. Good catch. I updated this in https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/135/commits/75868295f5ab1edce96613a67ff67244a483f303.

mrnosal commented:
I've added PilotPatientMTAge53.yml to the pilot-test-patients branch.

mickohanlon23 commented:
> I've added PilotPatientMTAge53.yml to the pilot-test-patients branch.

Patient MT53 now passes when I run the pilot test patients against this branch. That patient is now successfully getting a recommendation for a colposcopy from the errata.

mrnosal reviewed (approved):
IsMostRecentTest reviewed. Updated test pilot patients SJ42 and PN33 to match updated recommendations.

mickohanlon23 commented:
> IsMostRecentTest reviewed. Updated test pilot patients SJ42 and PN33 to match updated recommendations.

@mrnosal Thanks. I updated comments on the pilot partner's colpo clinic spreadsheet, as well as the one on our SharePoint made to keep track of the test cases. Updated comments indicate that the three patients called out above are now passing. I'll go ahead and merge this into `main`.
